### PR TITLE
Fix issue where DevTools would not be immediately available when using --start-paused

### LINKED ISSUE #126691: DevTools started late when running `flutter run` with `--start-paused`
DevTools is only started after the test passes:
Screenshot 2023-05-12 at 9 28 25 AM

Trying to click the vm service link before the test finishes yields this error message on redirect:
Screenshot 2023-05-12 at 9 22 14 AM

Clicking the vm service link after the test passes and the devtools uri is printed redirects to devtools successfully.

Running the same test with `flutter test` is WAI:
Screenshot 2023-05-12 at 9 57 02 AM

### PR DESCRIPTION
**Pull request by bkonyi**
Service extensions are unable to handle requests when the isolate they were registered on is paused. The DevTools launcher logic was waiting for some service extension invocations to complete before advertising the already active DevTools instance, but when --start-paused was provided these requests would never complete, preventing users from using DevTools to resume the paused isolate.

Fixes https://github.com/flutter/flutter/issues/126691

### CODE REVIEW

**Review by christopherfujino — COMMENTED** [review]
This looks reasonable, can you add a test?

**Comment by bkonyi** [comment]
> This looks reasonable, can you add a test?

That's the plan, I just wanted to make sure the existing checks didn't blow up first.

**Review by kenzieschmoll — APPROVED** [review]
LGTM, though we should find a way to test this so that it does not regress.

**Comment by bkonyi** [comment]
Sorry for the delay on this, totally forgot I had this draft. I've gone ahead and added a regression test so this is ready for review (FYI @christopherfujino).

**Review by christopherfujino — APPROVED** [review]
LGTM

File: packages/flutter_tools/lib/src/resident_devtools_handler.dart [hunk 1]
```diff
@@ -116,6 +112,10 @@ class FlutterResidentDevtoolsHandler implements ResidentDevtoolsHandler {
       // report their URLs yet. Do so now.
       _residentRunner.printDebuggerList(includeVmService: false);
     }
+
+    final List<FlutterDevice?> devicesWithExtension = await _devicesWithExtensions(flutterDevices);
```
Comment (bkonyi) [inline packages/flutter_tools/lib/src/resident_devtools_handler.dart]: Simply moving this seems to cause a race condition in `flutter error messages include a DevTools link` in `test/integration.shard/overall_experience_test.dart`. The test relies on the DevTools message to trigger a hot reload, which causes an overflow error that includes the DevTools URI. Unfortunately, the DevTools URI can only be output after the service extensions here have finished executing. Unless we want to put a delay in the test logic, we'll need to handle the `--start-paused` case separately here.

Comment (bkonyi) [reply packages/flutter_tools/lib/src/resident_devtools_handler.dart]: @christopherfujino can you take a look at the updated changes when you have a chance?

**Pull request merged**